cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -160,6 +160,9 @@ func Run(s *options.CMServer) error {
 	if err != nil {
 		return err
 	}
+	// it is more safe to use hostname plus pod name
+	// see https://github.com/kubernetes/kubernetes/issues/22775
+	id = id + "-" + os.Getenv("POD_NAME")
```

Comment:
In the config you changed, the pod is indeed running as a hostNetwork pod but is also a static pod and is not scheduled. The scenario you are trying to fix was predicated on the pod being scheduled.

If the scheduler is not running as a static pod, their is no reason for it to run on the hostNetwork. If you are running the scheduler in a real pod and on the hostNetwork your scenario where two schedulers are scheduled on the same node is *still* broken even after this change because there will be a port conflict between the two and one will crash loop. The fix is to not use  the host network when running the scheduler in a real pod (e.g. using a replica set).